### PR TITLE
Add S3UserClaim reconciliation logic

### PR DESCRIPTION
**Pull request by therealak12**
This pull request adds the S3UserClaim controller.
The controller tries to create/remove users in the Ceph cluster using S3UserClaim and S3User resources.

### CODE REVIEW

File: internal/controllers/s3userclaim/handler.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+*/
+
+package s3userclaim
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ceph/go-ceph/rgw/admin"
+	"github.com/go-logr/logr"
```
Comment (1995parham) [inline internal/controllers/s3userclaim/handler.go:24]: Using this logger is forced by kubernetes?

File: internal/controllers/s3userclaim/handler.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		ID:          "testuser",
+		DisplayName: "Test User",
+	})
+	if err != nil {
+		r.logger.Error(err, "failed to create ceph user")
+		return subreconciler.Requeue()
+	}
+
+	// Create subuser
+	if err := co.CreateSubuser(context.Background(), user, admin.SubuserSpec{
```
Comment (1995parham) [inline internal/controllers/s3userclaim/handler.go:124]: I think here we need to check that are we having `read` token or not. Am I right?

File: internal/controllers/s3userclaim/handler.go (hunk 1 above)
Comment (therealak12) [reply internal/controllers/s3userclaim/handler.go:26]: No. It uses `logr.Logger` which allows using any compatible implementations such as [`zapr`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/log/zap).

If you see advantages in using a different logging library, I'm happy to do so.

Comment (1995parham) [reply internal/controllers/s3userclaim/handler.go:26]: Awesome, we can use `zapr` in future :see_no_evil:

**Review by 1995parham — APPROVED** [review]

**Pull request merged**